core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -130,10 +141,12 @@ oppia.directive('searchBar', [function() {
 
         $scope.onSearchQueryChange = function(evt) {
           // Query immediately when the enter or space key is pressed.
-          if (evt.keyCode == 13 || evt.keyCode == 32) {
-            _onSearchQueryChangeExec();
+          if (evt.keyCode == 13) {
+            _onSearchQueryChangeExec(true);
+          } else if (evt.keyCode == 32) {
+            _onSearchQueryChangeExec(false);
           } else {
-            oppiaDebouncer.debounce(_onSearchQueryChangeExec, 400)();
+            oppiaDebouncer.debounce(_onSearchQueryChangeExec(false), 400)();
```

Comment:
Don't think you can do this -- look into JavaScript's 'bind' instead. Instead, what you're doing here is, calling _onSearchQueryChangeExec(false) and passing the return value (which is not a function) into debounce. This results in console errors.

Also, for simplicity -- how about we just go with false all the time, for now? Then we don't need the extra argument.